### ISSUE TITLE
New Check: UseWhileInsteadOfForLoopCheck

### ISSUE THREAD

**Issue by rdiachenko** [sevntu-checkstyle/sevntu.checkstyle#262]
**Statement:**

`while` loop should be used instead of `for` loop in the following cases:
- one or more parts (Initialization, Expression, Iteration) is/are omitted from the `for` loop declaration
- `for` loop declaration contains a complicated expression/logic
- ...

**Cause:**
- code is error prone and difficult to debug
- the code readability is very low
- ...

**E.g.:**
1)

```
// Actual:

for (; result != null && result.getType() != TokenTypes.SLIST
        && result.getType() != TokenTypes.OBJBLOCK; result = result.getParent()) {
    ...
}

// Expected:

while (result != null 
        && result.getType() != TokenTypes.SLIST
        && result.getType() != TokenTypes.OBJBLOCK) {
    ...
    result = result.getParent();
}
```

2)...

**Comment by damianszczepanik** [comment]
I agree with @romani that 'complicated' is not clear defined requirements but looking at first example I can imagine following rule.

Having:
```
for (START_SECTION; INCREMENTAL_SECTION; STOP_SECTION) {
    BODY_SECTION
}
````
if `START_SECTION` is empty we can always trigger and suggest to refactor to `while` loop as:
```
while (STOP_SECTION) {
    BODY_SECTION
    INCREMENTAL_SECTION
}
```

and that should be always possible and makes sense. It won't cover all cases but should not generate false-positive violations.
@rdiachenko What do you think?
